File: abc124_c.py
# C - Coloring Colorfully
# https://atcoder.jp/contests/abc124/tasks/abc124_c
"""
S = input()
N = len(S)
even = S[0::2]
odd = S[1::2]
ptn0 = N - (even.count('0') + odd.count('1'))
ptn1 = N - (even.count('1') + odd.count('0'))
print(min(ptn0, ptn1))
"""
S = input()
cnt = [[0,0],[0,0]]
for i, s in enumerate(S):
    x = int(i) % 2
    y = int(s)
    cnt[x][y] += 1
print(min(cnt[0][1] + cnt[1][0], cnt[0][0] + cnt[1][1]))
# if cnt[0][0] > cnt[1][0]:
#     print(cnt[0][1] + cnt[1][0])
# else:
#     print(cnt[0][0] + cnt[1][1])

File: atcoder/abc124_c.cpp
// C - Coloring Colorfully
// https://atcoder.jp/contests/abc124/tasks/abc124_c

#include <iostream>
#include <algorithm>
#include <string>
using namespace std;

int main() {
	string S;
	cin >> S;

	//初期値は大きい値
	int ans = S.size();

	//iは最初に使う色（0 or 1）
	for (int i = 0; i < 2; i++)
	{
		// iを最初に使う時に変えなければならないタイルの数
		int cnt = 0;

		for (int j = 0; j < S.size(); j++)
		{
			// (char)(0 + '0') -> '0';
			// (char)(1 + '0') -> '1';
			if ((j % 2 == 0) ^ (S[j] != (char)(i + '0'))) cnt++;
		}

		ans = min(ans, cnt);
	}

	cout << ans << endl;
	return 0;
}

// #include <iostream>
// #include <algorithm>
// #include <string>
// #include <vector>
// #include <tuple>
// #include <queue>
// // #include <bits/stdc++.h>
// using namespace std;

// int main() {
// 	string S;
// 	cin >> S;

// 	int N = S.size();
// 	int even0odd1 = 0, odd0even1 = 0;

// 	for (int i = 0; i < N; i++) {
// 		if (i % 2 == 0) {
// 			if (S.at(i) == '0')	even0odd1++;
// 			else                odd0even1++;
// 		}
// 		else {
// 			if (S.at(i) == '1')	even0odd1++;
// 			else                odd0even1++;
// 		}
// 	}

// 	cout << min(even0odd1, odd0even1) << endl;
// 	return 0;
// }
